python/dgl/data/csv_dataset.py

```diff
@@ -0,0 +1,339 @@
+import os
+import yaml
+from yaml.loader import SafeLoader
+import pandas as pd
+import numpy as np
+from .dgl_dataset import DGLDataset
+from ..convert import heterograph as dgl_heterograph
+from .. import backend as F
+from .utils import save_graphs, load_graphs, save_info, load_info
+
+
+def _yaml_sanity_check(meta_yaml):
+    #TODO: should be improved to check in an elegant way
+    with open(meta_yaml) as f:
```

Comment:
For configurations, you may want to take a look on pydantic https://pydantic-docs.helpmanual.io/, which is designed for config parsing and validation.

Reply:
nice module. I had a try. pls take a look.